src/QirTools/qirlib/src/interop/pyjit/intrinsics.rs

```diff
@@ -0,0 +1,237 @@
+#![allow(non_upper_case_globals)]
+#![allow(non_camel_case_types)]
+#![allow(non_snake_case)]
+#![allow(dead_code)]
+#![allow(unused_variables)]
+#![allow(unused_imports)]
+
+use microsoft_quantum_qir_runtime_sys::{PauliId, QirArray, QirRTuple, QirRuntime, QUBIT};
+use mut_static::ForceSomeRwLockWriteGuard;
+use std::ops::DerefMut;
+
+use super::gates::BaseProfile;
+
+fn get_current_gate_processor() -> ForceSomeRwLockWriteGuard<'static, BaseProfile> {
+    let v = crate::interop::pyjit::gates::CURRENT_GATES.write().unwrap();
+    v
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn __quantum__qis__h__body(qubit: QUBIT) {
+    log::debug!("/__quantum__qis__h__body/");
+    let mut gs = get_current_gate_processor();
+    gs.h(qubit);
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn __quantum__qis__h__ctl(ctls: *mut QirArray, qubit: QUBIT) {
+    log::debug!("/__quantum__qis__h__ctl/");
+    let control = get_qubit_id(ctls);
+    //let mut gs = get_current_gate_processor();
+    todo!("Not yet implemented.");
+    //gs.h_ctl(control, get_cubit_string(qubit));
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn __quantum__qis__measure__body(
+    qubits: *mut QirArray,
+    registers: *mut QirArray,
+) {
+    log::debug!("/__quantum__qis__measure__body/");
+
+    // get_qubit_id may return something like 94420488984834
+    // which will use up all computer memory
+
+    // let qubit = get_qubit_id(qubits);
+    // let mut gs = get_current_gate_processor();
+    // gs.m(qubit);
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn __quantum__qis__r__body(pauli: PauliId, theta: f64, qubit: QUBIT) {
+    log::debug!("/__quantum__qis__r__body/");
+    let mut gs = get_current_gate_processor();
+    match pauli {
+        1 => gs.rx(theta, qubit),
+        3 => gs.ry(theta, qubit),
+        2 => gs.rz(theta, qubit),
+        _ => panic!("Unsupported Pauli value: {}", pauli),
```

Comment:
Technically, rotating about `PauliI` is valid. It's a no-op in hardware, but still is valid.